src/Mutant.php

```diff
@@ -48,6 +49,11 @@ class Mutant
      */
     protected $process;
 
+    /**
+     * @var MutantResult
+     */
+    protected $result;
```

Comment:
Should Mutant preserve result in it?
Mutant is just a some kind of change that should preserve original file and modified file in it. 
Other stuff like process, container, or event tests that should be run to see when running mutation fails should be delegated to other classes IMO.
Adding here result increases responsibility of this class, and causes that mutant knows to much.
Look at: https://github.com/padraic/humbug/pull/122/files#diff-b6c9db5697c2a82bdfff464b8a7c2255R112

To setup result you have to put in there three parameters like $container, $group, $tracker...
What is group and tracker in Mutant context?

This change causes that whitout knowing those three things mutant cannot serve result.